src/rules/tsUtils.ts

```diff
@@ -320,6 +320,185 @@ export const isExpectCallWithParent = (
+  // isNamed<PossibleName extends Matcher>(
+  //   // name: PossibleName,
+  //   ...names: [PossibleName, ...PossibleName[]]
+  // ): this is ParsedExpectMatcher<PossibleName, Node>;
+}
+
+type BaseParsedModifier<
+  Modifier extends ModifierName = ModifierName
+> = ParsedExpectMember<Modifier>;
+
+type NegatableModifierName = ModifierName.rejects | ModifierName.resolves;
+type NotNegatableModifierName = ModifierName.not;
+
+/**
+ * Represents a parsed modifier that can be followed by a `not` negation modifier.
+ */
+interface NegatableParsedModifier<
+  Modifier extends NegatableModifierName = NegatableModifierName
+> extends BaseParsedModifier<Modifier> {
+  negation?: ExpectMember<ModifierName.not>;
+}
+
+/**
+ * Represents a parsed modifier that cannot be followed by a `not` negation modifier.
+ */
+export interface NotNegatableParsedModifier<
+  Modifier extends NotNegatableModifierName = NotNegatableModifierName
+> extends BaseParsedModifier<Modifier> {
+  negation?: never;
+}
+
+type ParsedExpectModifier =
+  | NotNegatableParsedModifier<NotNegatableModifierName>
+  | NegatableParsedModifier<NegatableModifierName>;
+
+interface Expectation<ExpectNode extends ExpectCall = ExpectCall> {
+  expect: ExpectNode;
+  modifier?: ParsedExpectModifier;
+  matcher?: ParsedExpectMatcher;
+}
+
+const parseExpectMember = <S extends ExpectPropertyName>(
+  expectMember: ExpectMember<S>,
+): ParsedExpectMember<S> => ({
+  name: getAccessorValue<S>(expectMember.property),
+  node: expectMember,
+});
+
+const reparseAsMatcher = (
+  parsedMember: ParsedExpectMember,
+): ParsedExpectMatcher => ({
+  ...parsedMember,
+  /**
+   * The arguments being passed to this `Matcher`, if any.
+   *
+   * If this matcher isn't called, this will be `null`.
+   */
+  arguments:
+    /* istanbul ignore next */
+    parsedMember.node.parent &&
+    parsedMember.node.parent.type === AST_NODE_TYPES.CallExpression
+      ? parsedMember.node.parent.arguments
+      : null,
+  // isNamed(...names: string[]) {
+  //   return names.includes(this.name);
+  // },
+});
+
+/**
+ * Re-parses the given `parsedMember` as a `ParsedExpectModifier`.
+ *
+ * If the given `parsedMember` does not have a `name` of a valid `Modifier`,
+ * an exception will be thrown.
+ *
+ * @param {ParsedExpectMember<ModifierName>} parsedMember
+ *
+ * @return {ParsedExpectModifier}
+ */
+const reparseMemberAsModifier = (
+  parsedMember: ParsedExpectMember<ModifierName>,
+): ParsedExpectModifier => {
+  if (isSpecificMember(parsedMember, ModifierName.not)) {
+    return parsedMember;
+  }
+
+  /* istanbul ignore if */
+  if (
+    !isSpecificMember(parsedMember, ModifierName.resolves) &&
+    !isSpecificMember(parsedMember, ModifierName.rejects)
+  ) {
+    throw new Error(
+      `modifier name must be either "${ModifierName.resolves}" or "${ModifierName.rejects}" (got "${parsedMember.name}")`,
+    ); // ts doesn't think that the ModifierName.not check is the direct inverse as the above two checks
```

Comment:
can you move these comments above the `throw` rather than inline on the closing braces?